src/components/SignUp/CareerInfo.js

```diff
@@ -74,6 +75,20 @@ class CareerInfo extends Component {
       Actions.refresh({ rightTitle: 'SAVE', onRight: this.onNextBtnPressed.bind(this) });
   }
 
+  componentWillReceiveProps(props) {
+    if(props.fromEdit && this.state.needRefresh) {
```

Comment:
my bad. you are right, you don't need to handle separately the case where it does not have fromEdit property. and here is the reference regarding prop validation: https://facebook.github.io/react/docs/typechecking-with-proptypes.html#react.proptypes